### PR TITLE
chore: raise test coverage to 80 percent

### LINKED ISSUE #188: Framework : Increase test coverage to 80%

### PR DESCRIPTION
**Pull request by jlouvel**
## Related Issue

Closes #188

---

## What does this PR do?

Increases JaCoCo test coverage from **67.47% to 80.01%** (instruction-level) by adding targeted branch and edge-case tests across high-miss classes.

**Approach:** iteratively identified the classes with the most uncovered instructions via `jacoco.csv`, wrote focused tests for guard clauses, error paths, and fallback branches, verified each batch with the full suite, and repeated until the 80% threshold was crossed.

### Coverage results

| Metric | Before | After |
|---|---|---|
| Instruction | 67.47% | 80.01% |
| Line | 68.87% | 80.34% |
| Branch | 54.31% | 69.34% |

---

## Tests

**New test classes (6):**
- `OperationStepExecutorBranchTest` — orchestration private/branch coverage
- `JettyStreamableHandlerTest` — MCP HTTP transport POST/GET/DELETE behavior
- `ProtocolDispatcherCoverageTest` — protocol error envelopes and edge cases
- `ResourceHandlerDynamicTest` — dynamic resource execution and fallback mapping
- `InputParameterSerializerTest` — input serializer nested/property branches
- `OutputParameterSerializerTest` — output serializer mapping/value semantics

**Expanded test classes (5):**
- `ConverterTest` — format guard clauses, JSONPath root selectors, maxLength edge cases
- `ResolverTest` — resolver and mapping branch paths
- `OperationStepExecutorIntegrationTest` — invalid call/mode paths
- `ResourceHandlerSafetyTest` — static-resource safety and template edge cases
- `ResourceRestletTest` — REST response/fallback behavior, forward-mode header propagation

**Validation:** `mvn test --no-transfer-progress` → 328 tests, 0 failures, 0 errors, 2 skipped, BUILD SUCCESS

---

## Checklist

- [x] CI is green (build, tests, schema validation, security scans)
- [x] Rebased on latest `main`
- [x] Small and focused — one concern per PR
- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/)

---

## Agent Context (optional)

```yaml
agent_name: GitHub Copilot
llm: Claude Opus 4.6
tool: VS Code Copilot Chat
confidence: high
source_event: "Issue #188 — increase test coverage to 80%"
discovery_method: code_review
review_focus: src/test/java/io/naftiko/**

### CODE REVIEW

File: src/test/java/io/naftiko/engine/exposes/rest/ResourceRestletTest.java [hunk 1]
```diff
@@ -126,6 +137,314 @@ public void handleShouldMapOutputParametersFromClientResponse() throws Exception
...
+        Capability capability = capabilityFromYaml(minimalCapabilityYaml());
+        RestServerSpec serverSpec = (RestServerSpec) capability.getServerAdapters().get(0)
+                .getSpec();
+        ResourceRestlet restlet = new ResourceRestlet(capability, serverSpec,
+                serverSpec.getResources().get(0));
+
+        java.lang.reflect.Method canBuildMockResponse = ResourceRestlet.class
+                .getDeclaredMethod("canBuildMockResponse", RestServerOperationSpec.class);
+        canBuildMockResponse.setAccessible(true);
+
```
Comment (eskenazit) [inline src/test/java/io/naftiko/engine/exposes/rest/ResourceRestletTest.java]: reflection call

File: src/test/java/io/naftiko/engine/exposes/rest/ResourceRestletTest.java [hunk 2]
```diff
@@ -126,6 +137,314 @@ public void handleShouldMapOutputParametersFromClientResponse() throws Exception
...
+        Capability capability = capabilityFromYaml(minimalCapabilityYaml());
+        RestServerSpec serverSpec = (RestServerSpec) capability.getServerAdapters().get(0)
+                .getSpec();
+        ResourceRestlet restlet = new ResourceRestlet(capability, serverSpec,
+                serverSpec.getResources().get(0));
+
+        java.lang.reflect.Method inOrDefault = ResourceRestlet.class
+                .getDeclaredMethod("inOrDefault", OutputParameterSpec.class);
+        inOrDefault.setAccessible(true);
+
```
Comment (eskenazit) [inline src/test/java/io/naftiko/engine/exposes/rest/ResourceRestletTest.java]: reflection call

File: src/test/java/io/naftiko/engine/exposes/OperationStepExecutorBranchTest.java [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+                          operations:
+                            - method: "GET"
+                              name: "x"
+                  consumes: []
+                """);
+
+        java.lang.reflect.Method addStepOutputToParameters = OperationStepExecutor.class.getDeclaredMethod(
+                "addStepOutputToParameters", Map.class, String.class, JsonNode.class);
+        addStepOutputToParameters.setAccessible(true);
+
```
[the remark this answers is not in the document]
Comment (jlouvel) [reply src/test/java/io/naftiko/engine/exposes/OperationStepExecutorBranchTest.java]: fixed

File: src/test/java/io/naftiko/engine/exposes/rest/ResourceRestletTest.java (hunk 1 above)
Comment (jlouvel) [reply src/test/java/io/naftiko/engine/exposes/rest/ResourceRestletTest.java]: fixed

File: src/test/java/io/naftiko/engine/exposes/rest/ResourceRestletTest.java [hunk 4]
```diff
@@ -126,6 +137,314 @@ public void handleShouldMapOutputParametersFromClientResponse() throws Exception
...
+        RestServerSpec serverSpec = (RestServerSpec) capability.getServerAdapters().get(0)
+                .getSpec();
+        ResourceRestlet restlet = new ResourceRestlet(capability, serverSpec,
+                serverSpec.getResources().get(0));
+
+        java.lang.reflect.Method buildParameterValue = ResourceRestlet.class
+                .getDeclaredMethod("buildParameterValue", OutputParameterSpec.class,
+                        ObjectMapper.class);
+        buildParameterValue.setAccessible(true);
+
```
[the remark this answers is not in the document]
Comment (jlouvel) [reply src/test/java/io/naftiko/engine/exposes/rest/ResourceRestletTest.java]: fixed
